integration/test/plan/README.md

```diff
@@ -0,0 +1,138 @@
+the Runtime.  The next block sets the GEOPM_PROFILE,
+GEOPM_PROGRAM_FILTER, and LD_PRELOAD variables appropriately for this
+app configuration.
+
+The next block sets up the configuration for the GEOPM Controller.
+The configuration enables summary reports, time-series traces, and
+profile traces to be emitted at the conclusion of the run.
+GEOPM_REPORT_SIGNALS adds additional signals to the report files with
+the given configuration.  GEOPM_NUM_PROC informs the Controller how
+many user processes it should expect to profile, and GEOPM_CTL_LOCAL
+tells the Controller not to use any MPI communication for transmitting
+data (assuming the Runtime was compiled with MPI support, which is the
+default).
+
+Prepending the invocation of `geopmctl` with `setsid` is vital to ensure
+that the save/restore mechanism of the Service is invoked.  This is
+required because of the session tracking design of the Service.  If
+you issue `geopmctl` without the `setsid`, any control changes that are
+made during the run will persist afterward until the current tty
+ends.
+
+Once the Controller is up and waiting for processes to connect, all
+that remains is to launch the desired app.  In this case, `numactl` is
+used to run `geopmbench` on the first package, and run `stress-ng` on the
+second package.  This script assumes the system has dual-socket
+CPUs.
+
+The `wait` call at the end ensures that the script does not return until
+all apps are complete.
+
+For more information on the environment variables used to configure
+GEOPM see: [man 7 geopm](https://geopm.github.io/geopm.7.html).
+
+When the script is complete several GEOPM output files will be
+present in the current working directory, suffixed with the compute
+node that they were executed on: the summarized report file -
+`test_multi_app_report.yaml-node1`, the time series data -
+`test_multi_app_trace.csv-node1`, and the time series data of
+per-process messages between the application and the Controller -
+`test_multi_app_trace_profile.csv-node1`.  These files will be used to
+verify the run completed as expected in the next section.
+
+### test_multi_app.py
+This file contains the test harness necessary to launch
+`test_multi_app.sh` and also the test assertions that verify the data
+gathered during the run.
+
+By default, this test is setup to run on a single compute node.  After
+`test_multi_app.sh` completes, the following assertions are tested:
+
+1. `test_meta_data` - The test ran on the expected number of nodes.
+2. `test_expected_regions_exist` - The summarized report file contains
+   data for the expected set of application regions.
+   - Notice that `MPI_Init_thread` is still listed as an expected
+     region even though the application does not use MPI.  This is
+     because the GEOPM Runtime is still built with MPI support by
+     default, and thus will initialize MPI even if the application
+     will not use it.  In this case, `geopmbench` *is* being invoked
```

Comment:
```suggestion
     region even though the application does not use MPI.  This is
     because the GEOPM Runtime is still built with MPI support by
     default, and will track calls to MPI_Init_thread by any application
     process that calls it.  In this case, `geopmbench` *is* being invoked
```